dotCMS/src/integration-test/java/com/dotcms/variant/VariantAPITest.java

```diff
@@ -0,0 +1,340 @@
+
+        ArrayList results = getResults(variant);
+        assertFalse(results.isEmpty());
+
+        APILocator.getVariantAPI().delete(variant.identifier());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#get(String)}
+     * When: Try to get  {@link Variant} by id
+     * Should: get it
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void get() throws DotDataException {
+        final Variant variant = new VariantDataGen().nextPersisted();
+
+        ArrayList results = getResults(variant);
+        assertFalse(results.isEmpty());
+
+        final Optional<Variant> variantFromDataBase = APILocator.getVariantAPI().get(variant.identifier());
+
+        assertTrue(variantFromDataBase.isPresent());
+        assertEquals(variant.identifier(), variantFromDataBase.get().identifier());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#get(String)}
+     * When: Try to get  {@link Variant} by name
+     * Should: get it
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void getByName() throws DotDataException {
+        final Variant variant = new VariantDataGen().nextPersisted();
+
+        ArrayList results = getResults(variant);
+        assertFalse(results.isEmpty());
+
+        final Optional<Variant> variantFromDataBase = APILocator.getVariantAPI().getByName(variant.name());
+
+        assertTrue(variantFromDataBase.isPresent());
+        assertEquals(variant.identifier(), variantFromDataBase.get().identifier());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#get(String)}
+     * When: Try to get  {@link Variant} by id that not exists
+     * Should: return a {@link Optional#empty()}
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void getNotExists() throws DotDataException {
+
+        final Optional<Variant> variantFromDataBase = APILocator.getVariantAPI()
+                .get("Not_Exists");
+
+        assertFalse(variantFromDataBase.isPresent());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#get(String)}
+     * When: Try to get  {@link Variant} by id equals to NULL
+     * Should:
+     *
+     * @throws DotDataException
+     */
+    @Test(expected = NullPointerException.class)
+    public void getWithNull() {
+        APILocator.getVariantAPI().get(null);
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#get(String)}
+     * When: Try to get  {@link Variant} by id that not exists
+     * Should: return a {@link Optional#empty()}
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void getByNameNotExists() {
+
+        final Optional<Variant> variantFromDataBase = APILocator.getVariantAPI()
+                .getByName("Not_Exists");
+
+        assertFalse(variantFromDataBase.isPresent());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#get(String)}
+     * When: Try to get  {@link Variant} by id equals to NULL
+     * Should:
```

Comment:
Should missing

Reply:
done https://github.com/dotCMS/core/pull/22839/commits/f5ec6589e6b865f668a1d3b0085035ddbb35b6a6